compiler/src/dmd/init.d

```diff
@@ -167,6 +167,7 @@ extern (C++) final class ArrayInitializer : Initializer
     uint dim;               // length of array being initialized
     Type type;              // type that array will be used to initialize
     bool sem;               // true if semantic() is run
+    bool isCarray;          // C array semantics
```

Comment:
Will need this adding to the headers too so that gdc/ldc can get to it in their codegen.

Reply:
done